checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+   */
+  private void doGenKillForPseudoAssignment(
+      Node node,
+      Set<ImmutableSet<LocalVarWithTree>> obligations,
+      LocalVariableNode lhsVar,
+      Node rhs) {
+    // Replacements to eventually perform in obligations.  We keep this map to avoid a
+    // ConcurrentModificationException in the loop below.
+    Map<ImmutableSet<LocalVarWithTree>, ImmutableSet<LocalVarWithTree>> replacements =
+        new LinkedHashMap<>();
+    // construct lhsVarWithTreeToGen once outside the loop for efficiency
+    LocalVarWithTree lhsVarWithTreeToGen =
+        new LocalVarWithTree(new LocalVariable(lhsVar), node.getTree());
+    for (ImmutableSet<LocalVarWithTree> obligation : obligations) {
+      Set<LocalVarWithTree> kill = new LinkedHashSet<>();
+      // always kill the lhs var if present
+      addLocalVarWithTreeToSetIfPresent(obligation, lhsVar.getElement(), kill);
+      LocalVarWithTree gen = null;
+      // if rhs is a variable tracked in the obligation's resource alias set, gen the lhs
+      if (rhs instanceof LocalVariableNode) {
+        LocalVariableNode rhsVar = (LocalVariableNode) rhs;
+        for (LocalVarWithTree lvwt : obligation) {
+          if (lvwt.localVar.getElement().equals(rhsVar.getElement())) {
+            gen = lhsVarWithTreeToGen;
+            // We remove temp vars from tracking once they are assigned to another location.
+            if (typeFactory.isTempVar(rhsVar)) {
+              addLocalVarWithTreeToSetIfPresent(obligation, rhsVar.getElement(), kill);
+            }
+            break;
+          }
+        }
+      }
+      // Check if there is something to do before creating a new obligation, for efficiency.
+      if (kill.isEmpty() && gen == null) {
+        continue;
+      }
+      Set<LocalVarWithTree> newObligation = new LinkedHashSet<>(obligation);
+      newObligation.removeAll(kill);
+      if (gen != null) {
+        newObligation.add(gen);
+      }
+      if (newObligation.size() == 0) {
+        // We have killed the last reference to the resource; check the must-call obligation.
+        MustCallAnnotatedTypeFactory mcAtf =
+            typeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+        checkMustCall(
+            obligation,
+            typeFactory.getStoreBefore(node),
+            mcAtf.getStoreBefore(node),
+            "variable overwritten by assignment " + node.getTree());
+      }
+      replacements.put(obligation, ImmutableSet.copyOf(newObligation));
+    }
+    // Finally, update obligations according to the replacements.
+    for (Map.Entry<ImmutableSet<LocalVarWithTree>, ImmutableSet<LocalVarWithTree>> entry :
+        replacements.entrySet()) {
+      obligations.remove(entry.getKey());
+      if (!entry.getValue().isEmpty()) {
+        obligations.add(entry.getValue());
+      }
+    }
+  }
+
+  /**
+   * If a {@link LocalVarWithTree} is present in {@code resourceAliasSet} whose variable element is
```

Comment:
"a LocalVarWithTree":  this method actually does them all.  Should this be modified, or should there be a `break` or `return` statement inside the `if` body?